### ISSUE TITLE
Can't access SignalR logging on Android

### ISSUE THREAD

**Issue by lepicekmichal** [dotnet/aspnetcore#40392]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Logger is not preffered solution for android and even if we use it somehow internally, then we cannot turn it on. That's because it uses HubConnection class itself for tag. And that gets proguarded/obfuscated. 
ADB does not allow tag of asterisk or dots - eg: "*.*.*.a$HubConnection".

Please allow us to set up logging from java/kotlin side on Android.

### Expected Behavior

Logging into Logcat and possible override.

**Comment by rafikiassumani-msft** [comment]
@lepicekmichal Are you running an obfuscator on the SignalR library? In case this is the issue, and you would like to hardcode the tag name, then we will welcome a PR.

**Comment by lepicekmichal** [comment]
I don't run obfuscator, I guess I looked wrong. Nevertheless LoggerFactory.getLogger(Class<*> class) does return malformed tag and on top of that, it's not extendable like the other platforms.
Why is HubConnectionBuilder missing _configureLogging_ method?

**Comment by BrennanConroy** [comment]
There is no `configureLogging` method because Java configures logging globally for your app. We use slf4j and you bring your own logging system that plugs into slf4j
https://docs.microsoft.com/aspnet/core/signalr/java-client?view=aspnetcore-6.0#add-logging

**Comment by lepicekmichal** [comment]
I saw these docs, unfortunately it doesn't print anything, at all. I still think it's a bug, but if someone can elaborate on how to make it work, extend the documentation chapter, that would be greatly appreciated.

**Comment by BrennanConroy** [comment]
What are you trying that isn't working?

**Comment by lepicekmichal** [comment]
Well I added dependencies, either of these:
```
implementation "org.slf4j:slf4j-jdk14:1.7.36"
implementation 'org.slf4j:slf4j-android:1.7.36'
```

And that's pretty much it, there is no extensibility.

I printed out the name of logger which is provided inside HubConnection class like this `LoggerFactory.getLogger(HubConnection.class);`; and the name it printed out was something like _*.*.a$HubConnection_.
So there was no way I could turn on that tag via adb shell prop.

I could also try other 3rd party dependencies which have docs about how to add configuration file making it possible to print inside logcat, but that would not help with extensibility. Can't show them on screen, can't upload them to our log servers.

**Comment by BrennanConroy** [comment]
> I printed out the name of logger which is provided inside HubConnection class like this `LoggerFactory.getLogger(HubConnection.class);`; and the name it printed out was something like _*.*.a$HubConnection_.

If you manually create a logger with "HubConnection" as the name does that work in your scenario?

That's what we were considering in https://github.com/dotnet/aspnetcore/issues/40392#issuecomment-1050191417

**Comment by lepicekmichal** [comment]
Yeah, if it got replaced inside library with such string, it would help with not seeing any logs at all. 
Would not help with processing the logs ourselves though.

**Comment by rafikiassumani-msft** [comment]
@lepicekmichal We are not planning to implement `ConfigureLogging` to allow users to pass in a `Logger` from outside. Normally a `Logger` applies to a class and works within the class context for tagging reasons.  It does not make sense to keep passing a single instance of `HubConnection` logger from `ConfigureLogging` to other classes as you'll end up losing the logs context due to a single tag. 

We are, however, planning to change SignalR to start using a simple string for log tagging rather than the full package name plus class name.  

We will change all the loggers to use the following format: ` Logger = LoggerFactory.getLogger("SignalR.<ClassName>")`.  Example for HubConnection class: ` Logger = LoggerFactory.getLogger("SignalR.HubConnection")` We hope that proguard/obfuscator will not mess up with that string and you should be able to process logs with `SignalR.<ClassName>` prefix. Let us know if this will work for you.

**Comment by lepicekmichal** [comment]
Yes, that would help with debugging as of right now there are no logs whatsoever.
As for published version suppose I should be able to find slfj4 solution.

One more thing, this way we will not now which hub is logging what. There should be possibility to pass additional tag info. Or maybe you can just append url to the text of the log, something like that.
